python_fundamentals-master/07_classes_objects_methods/07_01_car.py

```diff
@@ -10,20 +10,26 @@
 
 '''
 
-class car():
-    def __init__(self, model, year, max_speed):
+class Car():
+    def __init__(self, model, year, speed):
         self.model = model
         self.year = year
-        self.max_speed = int(max_speed)
-
-    def speed_up(self, speed):
-        print("max_speed is " + (int(speed) + int(self.max_speed))
-
-smart = car("smart", "1993", 210)
-print(smart.model)
-print(smart.year)
-print(smart.speed_up(5))
-volvo = car("volvo", "2010", 260)
-print(volvo.model)
-print(volvo.year)
-print(volvo.speed_up(5))
+        self.speed = speed
+    def speed_up(self, speed_up, accelate):
+        self.accelarate = 5
+        self.speed_up = (self.speed + self.accelarate)
+        return f"max_speed is {self.speed_up}"
```

Comment:
So i would not add a class attribute for accelerate, simply because, is this something that needs to be remembered? Each time it may be different right?  Also, you don't need a new class attribute `self.speed_up` this is not something you need to remember either.
In fact, you just need to _modify_ the current speed, which is stored in `self.speed`.

Also, you are hardcoding it instead of using the variable you passed in the function. 
You can just use the variable directly to modify the speed

So I think you have made this more complicated than you need because you are thinking that you need to create variables, but you don't.

```
def speed_up(self, accelerate):
    self.speed += accelerate
    return "message here"
```
this is all that's really needed. You are just need to modify the speed of the car. You can do that with the argument passed into the `speed_up` (accelerate)
hope this makes sense.

Reply:
Thanks for the example. It is good to remember that the attributes is information that you can use inside a function. Just sometimes the attributes stay grey, and I get confused and want to make it a variable.  But this makes a lot of sense

Reply:
so in this current code you still have the problem I mentioned in the other pull request.  Line 20, overwrites the method `speed_up` and turns it into an integer